src/main/java/se/sundsvall/byggrintegrator/configuration/ByggrFileExceptionConfig.java

```diff
@@ -0,0 +1,103 @@
+package se.sundsvall.byggrintegrator.configuration;
+
+import java.util.Locale;
+import java.util.Optional;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.bind.annotation.ResponseBody;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.thymeleaf.ITemplateEngine;
+import org.thymeleaf.context.Context;
+import org.zalando.problem.Problem;
+import org.zalando.problem.Status;
+import org.zalando.problem.StatusType;
+import org.zalando.problem.ThrowableProblem;
+
+import se.sundsvall.byggrintegrator.api.ByggrFileListResource;
+import se.sundsvall.dept44.requestid.RequestId;
+
+import jakarta.validation.ValidationException;
+
+/**
+ * Configuration to convert exceptions and problems to HTML responses.
+ */
+@Configuration
+public class ByggrFileExceptionConfig {
+
+	private static final Logger LOG = LoggerFactory.getLogger(ByggrFileExceptionConfig.class);
+
+	@ControllerAdvice(assignableTypes = ByggrFileListResource.class)
+	public static class ByggrFileExceptionAsHtmlHandler {
```

Comment:
Rename to ControllerExceptionAsHtmlHander (as it handles several types of exceptions and not just ByggrFileExceptions)?